### PR TITLE
Verified derivative methods

### PR DESCRIPTION
**Pull request by dloney**
Full build of initial LM engine upgrade

### CODE REVIEW

File: include/LevenbergAlgorithm2.h [hunk 1]
```diff
@@ -96,6 +96,7 @@ class LevenbergAlgorithm : public Algorithm {
       double m_StepToleranceMaximum = 0;
 
       int m_LambdasMinimum = 1;
+      int m_DerivativeType = 2;
```
Comment (julemai) [inline include/LevenbergAlgorithm2.h:99]: Maybe add in comment which Type=2 actually is (forward/backward/central)?

**Review by julemai — APPROVED** [review]
I am not super deep into the algorithm itself. But what I understand is looking good to me.

**Review by lsmatott — APPROVED** [review]
Looks good to me.

File: include/LevenbergAlgorithm2.h (hunk 1 above)
Comment (dloney) [reply include/LevenbergAlgorithm2.h]: Good idea. I forgot which that was by the time I went back and looked. I fixed this in the changes I just made.

**Comment by dloney** [comment]
There was an issue with the passing of tied parameters across the MPI interface, namely I had forgotten to append those to the vector to be transferred. This recent commit fixes that. I'm testing on our hydrologic model with the most parameters, almost 10,000. I haven't seen any performance issues. Memory is a bit higher than I would like, but it's reasonable with this number of parameters and alternatives being stored and processed.

**Pull request merged**